clar2wasm/src/wasm_generator.rs

```diff
@@ -489,6 +489,36 @@ impl ASTVisitor for WasmGenerator {
                 .funcs
                 .by_name(&format!("sqrti-{type_suffix}"))
                 .unwrap_or_else(|| panic!("function not found: sqrti-{type_suffix}")),
+            NativeFunctions::BitwiseAnd => self
+                .module
+                .funcs
+                .by_name(&format!("bit-and-{type_suffix}"))
+                .unwrap_or_else(|| panic!("function not found: bit-and-{type_suffix}")),
+            NativeFunctions::BitwiseNot => self
+                .module
+                .funcs
+                .by_name(&format!("bit-not-{type_suffix}"))
+                .unwrap_or_else(|| panic!("function not found: bit-not-{type_suffix}")),
+            NativeFunctions::BitwiseOr => self
+                .module
+                .funcs
+                .by_name(&format!("bit-or-{type_suffix}"))
+                .unwrap_or_else(|| panic!("function not found: bit-or-{type_suffix}")),
+            NativeFunctions::BitwiseLShift => self
+                .module
+                .funcs
+                .by_name(&format!("bit-shift-left-{type_suffix}"))
+                .unwrap_or_else(|| panic!("function not found: bit-shift-left-{type_suffix}")),
+            NativeFunctions::BitwiseRShift => self
+                .module
+                .funcs
+                .by_name(&format!("bit-shift-right-{type_suffix}"))
+                .unwrap_or_else(|| panic!("function not found: bit-shift-right-{type_suffix}")),
+            NativeFunctions::BitwiseXor2 => self
+                .module
+                .funcs
+                .by_name(&format!("bit-xor-{type_suffix}"))
+                .unwrap_or_else(|| panic!("function not found: bit-xor-{type_suffix}")),
```

Comment:
This should not be declared here. Bitwise operators should be parsed by `traverse_bitwise`, not `traverse_arithmetic`.

To know which operators should be parsed by which function, have a look at `traverse_expr` in _ast\_visitor.rs_.

Reply:
oh, and there is a `traverse_bit_shift` apparently too! :o